lib/drivers/bayerContourNext.js

```diff
@@ -0,0 +1,501 @@
+    data.id = data.model + '-' + data.serialNumber;
+    cfg.builder.setDefaults({ deviceId: data.id});
+    var dataToPost = [];
+    if (data.bgmReadings.length > 0) {
+      for (var i = 0; i < data.bgmReadings.length; ++i) {
+        var datum = data.bgmReadings[i];
+        var smbg = cfg.builder.makeSMBG()
+          .with_value(datum.glucose)
+          .with_deviceTime(datum.displayTime)
+          .with_timezoneOffset(datum.displayOffset)
+          .with_time(datum.displayUtc)
+          .with_units(datum.units)
+          .set('index', datum.nrec)
+          .done();
+          if (datum.annotations) {
+            _.each(datum.annotations, function(ann) {
+              annotate.annotateEvent(smbg, ann);
+            });
+          }
+        dataToPost.push(smbg);
+      }
+    }else{
+      debug('Device has no records to upload');
+      throw(new Error('Device has no records to upload'));
+    }
+
+    return dataToPost;
+  };
+
+  return {
+    detect: function(deviceInfo, cb){
+      debug('no detect function needed', arguments);
+      cb(null, deviceInfo);
+    },
+
+    setup: function (deviceInfo, progress, cb) {
+      console.log('in setup!');
+      progress(100);
+      cb(null, {deviceInfo: deviceInfo});
+    },
+
+    connect: function (progress, data, cb) {
+      debug('in connect!');
+
+      cfg.deviceComms.connect(data.deviceInfo, bcnPacketHandler, probe, function(err) {
+        if (err) {
+          return cb(err);
+        }
+        data.disconnect = false;
+        progress(100);
+        cb(null, data);
+
+      });
+    },
+
+    getConfigInfo: function (progress, data, cb) {
+      debug('in getConfigInfo', data);
+
+      getDeviceInfo({}, function (err, result) {
+          progress(100);
+
+          if(!err){
+              data.connect = true;
+              _.assign(data, result);
+
+              cb(null, data);
+          }else{
+              cb(err,result);
+          }
+      });
+    },
+
+    fetchData: function (progress, data, cb) {
+      debug('in fetchData', data);
+
+      function getOneRecordWithProgress(recnum, cb) {
+        var rec = data.rawrecords.shift();
+        progress(100.0 * recnum / data.nrecs);
+        setTimeout(function() {
+          getOneRecord(rec, data, cb);
+        }, 20);
+      }
+
+      async.timesSeries(data.nrecs, getOneRecordWithProgress, function(err, result) {
+        if (err) {
+          debug('fetchData failed');
+          debug(err);
+          debug(result);
+        } else {
+          debug('fetchData', result);
+        }
+        data.fetchData = true;
+        data.bgmReadings = result;
+        progress(100);
+        cb(err, data);
+      });
+    },
+
+    processData: function (progress, data, cb) {
+        progress(0);
+        data.bg_data = processReadings(data.bgmReadings);
+        data.post_records = prepBGData(progress, data);
+        var ids = {};
+        for (var i = 0; i < data.post_records.length; ++i) {
```

Comment:
We have not seen duplicates either, but remember we only have two devices in our hand so difficult to assert, so based on the other drivers implementations available we thought it was a good idea to check it.

Regarding the logIndices and the comment, if we remove the index here you won't get the logIndices filled either, in objectBuilder, or I'm missing something?